### PR TITLE
Add ms-gsl

### PR DESCRIPTION
**Pull request by apmorton**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Package does not ship static libraries. If static libraries are needed, [follow CFEP-18](https://github.com/conda-forge/cfep/blob/main/cfep-18.md).
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by apmorton** [comment]
Willing to maintain

**Comment by apmorton** [comment]
@conda-forge/help-c-cpp, ready for review!

File: recipes/ms-gsl/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+
+test:
+  requires:
+    - cmake
+    - {{ compiler('cxx') }}
+    - ninja
+  files:
+    - test
+
+  commands:
```
Comment (loriab) [inline recipes/ms-gsl/meta.yaml:30]: It's encouraged to add some tests checking that the expected headers, libraries, executables are present in the package: https://github.com/conda-forge/zeromq-feedstock/blob/main/recipe/meta.yaml#L46-L64 . And if you are shipping headers, a `build/run_exports` is encouraged.

(P.S. I'm not an official reviewer, so above is just an observation.)

Comment (apmorton) [reply recipes/ms-gsl/meta.yaml:30]: The rest subdirectory of the recipe contains a fully formed program that uses the library and compiled against it.

Not sure I follow the run_exports bit, can you explain?

For context, this is a header only library and has no dependencies.

Comment (loriab) [reply recipes/ms-gsl/meta.yaml:30]: `run_exports` is a way for library recipe developers to impose that if a downstream package compiles against this one, that package should have this one in its requirements/run with certain constraints. https://docs.conda.io/projects/conda-build/en/latest/resources/define-metadata.html#export-runtime-requirements But if this is header-only, run_exports is irrelevant.

I agree, your test build largely assures the package is functioning. But I suspect the reviewers may like file checks anyways. Here's a header-only pattern. https://github.com/conda-forge/eigen-feedstock/blob/main/recipe/meta.yaml#L27-L30

Comment (carterbox) [inline recipes/ms-gsl/meta.yaml:30]:
```suggestion
  commands:
    - test -f ${PREFIX}/include/gsl/gsl  # [unix]
    - if not exist %PREFIX%\\Library\\include\\gsl\\gsl exit 1  # [win]
    - test -f $PREFIX/share/cmake/Microsoft.GSL/Microsoft.GSLConfig.cmake  # [unix]
    - if not exist %PREFIX%\\Library\\share\\cmake\\Microsoft.GSL\\Microsoft.GSLConfig.cmake exit 1 # [win]

```
These tests make it obvious that this is a header-only library by only looking at the recipe.

**Pull request merged**